webpack/scenes/ContentViews/Details/Histories/ContentViewHistories.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, shallowEqual } from 'react-redux';
+import { TableVariant } from '@patternfly/react-table';
+import { translate as __ } from 'foremanReact/common/I18n';
+import LongDateTime from 'foremanReact/components/common/dates/LongDateTime';
+import { STATUS } from 'foremanReact/constants';
+import PropTypes from 'prop-types';
+
+import TableWrapper from '../../../../components/Table/TableWrapper';
+import { getContentViewHistories } from '../ContentViewDetailActions';
+import {
+  selectCVHistories,
+  selectCVHistoriesStatus,
+  selectCVHistoriesError,
+} from '../ContentViewDetailSelectors';
+
+const ContentViewHistories = ({ cvId }) => {
+  const response = useSelector(state => selectCVHistories(state, cvId), shallowEqual);
+  const status = useSelector(state => selectCVHistoriesStatus(state, cvId), shallowEqual);
+  const error = useSelector(state => selectCVHistoriesError(state, cvId), shallowEqual);
+  const loading = status === STATUS.PENDING;
+
+  const [rows, setRows] = useState([]);
+  const [metadata, setMetadata] = useState({});
+  const [searchQuery, updateSearchQuery] = useState('');
+
+  const columnHeaders = [
+    __('Date'),
+    __('Version'),
+    __('Status'),
+    __('Action'),
+    __('Description'),
+    __('User'),
+  ];
+
+  const buildRows = (results) => {
+    const newRows = [];
+    results.forEach((history) => {
+      const {
+        version,
+        created_at: createdAt,
+        status: taskStatus,
+        desciption,
+        action,
+        user,
+      } = history;
+
+      const cells = [
+        { title: <LongDateTime date={createdAt} showRelativeTimeTooltip /> },
+        version,
+        taskStatus,
+        action,
+        desciption,
+        user,
+      ];
+
+      newRows.push({ cells });
+    });
+    return newRows;
+  };
+
+  useEffect(() => {
+    const { results, ...meta } = response;
+    setMetadata(meta);
+    if (!loading && results) {
+      const newRows = buildRows(results);
+      setRows(newRows);
+    }
+  }, [JSON.stringify(response)]);
+
+  const emptyContentTitle = __("You currently don't have any history for this content view.");
+  const emptyContentBody = __('You currently don\'t have any history for this content view.'); // needs link
```

Comment:
Might be helpful to include a message about how to get to a non-empty state, e.g. `History will appear here when the content view is published or promoted.`